kwok/cloudprovider/helpers.go

```diff
@@ -30,78 +32,91 @@ import (
 	"sigs.k8s.io/karpenter/pkg/scheduling"
 )
 
+var (
+	// AWS uses (family).(size) format
+	awsRegexp = regexp.MustCompile(`.*\.(small|medium|large|\d*xlarge)`)
+)
+
 type InstanceTypeOptions struct {
-	Name               string
-	Offerings          cloudprovider.Offerings
-	Architecture       string
-	OperatingSystems   sets.Set[string]
-	Resources          v1.ResourceList
-	InstanceTypeLabels map[string]string
+	Name             string                  `json:"name"`
+	Offerings        cloudprovider.Offerings `json:"offerings"`
+	Architecture     string                  `json:"architecture"`
+	OperatingSystems sets.Set[string]        `json:"operatingSystems"`
+	Resources        v1.ResourceList         `json:"resources"`
+
+	// These are used for setting default requirements, they should not be used
+	// for setting arbitrary node labels.  Set the labels on the created NodePool for
+	// that use case.
+	instanceTypeLabels map[string]string
 }
 
-func MakeInstanceTypeLabels(cpu, memFactor int) map[string]string {
-	size := fmt.Sprintf("%dx", cpu)
-	var family string
-	switch memFactor {
-	case 2:
-		family = "c" // cpu
-	case 4:
-		family = "s" // standard
-	case 8:
-		family = "m" // memory
-	default:
-		family = "e" // exotic
+// ConstructInstanceTypes create many unique instance types with varying CPU/memory/architecture/OS/zone/capacity type.
+func ConstructInstanceTypes() ([]*cloudprovider.InstanceType, error) {
+	instanceTypeFile := os.Getenv(kwokInstanceTypeFileKey)
+	if instanceTypeFile == "" {
+		return nil, fmt.Errorf("no instance type file specified")
```

Comment:
Just checking, this is defaulted to the currently hard-coded file, right?

Reply:
no longer relevant since we're embedding